pkg/services/provisioning/dashboards/validator.go

```diff
@@ -0,0 +1,97 @@
+package dashboards
+
+import (
+	"context"
+	"time"
+
+	"github.com/grafana/grafana/pkg/infra/log"
+)
+
+type duplicate struct {
+	Sum             uint8
+	InvolvedReaders map[string]struct{}
+}
+
+func newDuplicate() *duplicate {
+	return &duplicate{InvolvedReaders: make(map[string]struct{})}
+}
+
+type duplicateEntries struct {
+	Titles map[dashboardIdentity]*duplicate
+	UIDs   map[string]*duplicate
+}
+
+func newDuplicateEntries() *duplicateEntries {
+	return &duplicateEntries{Titles: make(map[dashboardIdentity]*duplicate), UIDs: make(map[string]*duplicate)}
+}
```

Comment:
For simplicity, can we just remove this function and inline the call? It's only used in one place and I think inlining makes for less code.